### DISCUSSION TITLE
Empty stage 4 validation

### DISCUSSION THREAD

**Discussion by rkottmann** [OP-TED/eForms-SDK#327] [Q&A] [answered]
Hi,

I only see an empty pattern in SDK 1.5.1 

https://github.com/OP-TED/eForms-SDK/blob/1.5.1/schematrons/static/validation-stage-4.sch

Question 1: Is this a bug?
Question 2: If it is not a bug, where do I find the rules for conditional mandatory fields? The README.md pointed me to stage 4

thx in advance,

renzo

**Comment by bertrand-lorentz** [comment, marked as answer]
Hi,

The rules for conditional mandatory fields were removed in version 1.3.0 of the SDK, as described in the release notes: https://github.com/OP-TED/eForms-SDK/releases/tag/1.3.0

So it's not a bug, it was intentional, in order to make it initially easier to fill in notices for end users.
We will bring back at least some of those rules in future versions of the SDK, with improvements and updates.

If you wish to have an idea on how those rules were, you can have a look in SDK version 1.2.1.
